add docs for CI components (#110)
kubev2v/forkliftci: cluster/providers/README.md

# Migration source providers

## Installation
Provider can be installed on a K8s cluster :

`providers/install-provider.sh <provider-name>`

when `<provider-name>` can be either of:
- openstack
- ovirt
- vmware

## openstack 

> containerised [packstack](https://github.com/kubev2v/packstack-img)  deployment  that is tailor made for forklift  migrations.

Reviewer: ```suggestion
containerised [packstack](https://github.com/kubev2v/packstack-img)  deployment  that is tailored for forklift migrations.
```

- Environment variables:

    | Name | Description | Required | 
    | ----------- | ----------- | ---  |
    | NFS_IP_ADDRESS |  NFS server  address used by cinder | yes | 
    | NFS_SHARE |  NFS export path. | yes |
    | KEYSTONE_USE_SSL | deploy  Keystone with SSL and self signed certs  | no |

## ovirt 
[fakeovirt](https://github.com/kubev2v/fakeovirt) and [ovirt-imageio](https://github.com/kubev2v/ovirt-imageio-server) deployments :
- Environment variables:

    | Name | Description | Required | 
    | ----------- | ----------- | ---  |

>     | NFS_IP_ADDRESS |  NFS server  address used by cinder | yes | 

Reviewer: cinder?

Author: Yes we use NFS as [backend](https://github.com/kubev2v/packstack-img/blob/main/files/packstack.answer#L503C1-L503C1) for cinder . 

it is injected to the cinder nfs_shares mounts here https://github.com/kubev2v/forkliftci/blob/main/cluster/providers/openstack/utils.sh#L277 ,
also it used by the OSP volume populator which use csi-driver-nfs   in the backend.

Reviewer: ack

    | NFS_SHARE |  NFS export path. | yes |

> ## vmware

Reviewer: ```suggestion
## VMware
```

> vmware/vcsim  deployment  and vsphere-provider with stub vddkInitImage [image](../../stub-images/vddk-test-vmdk/BUILD.bazel).

Reviewer: ```suggestion
VMware/vcsim deployment and vsphere-provider with stub vddkInitImage [image](../../stub-images/vddk-test-vmdk/BUILD.bazel).
```
we can also add a link to https://github.com/vmware/govmomi/tree/main/vcsim
